src/main/webapp/go.js

```diff
@@ -198,14 +247,15 @@ function updateToCurrentState(index) {
   if (index <= -1) { // The user has not yet pressed the start button.
```

Comment:
isn't index the same as hunt.destIndex?  You may not need to pass this variable around if hunt contains the updated index